File: frontend/src/features/receipts/types/receipt.ts
export interface Receipt {
    id: string;
    user_id: string;
    vehicle_id: string;
    service_visit_id: string | null;

    original_filename: string;
    stored_filename: string;
    file_url: string;

    content_type: string;
    file_size: number;

    notes: string | null;

    created_at: string;
    updated_at: string;
}

export interface CreateReceiptPayload {
    vehicleId: string;
    file: File;
    serviceVisitId?: string;
    notes?: string;
}

export interface UpdateReceiptPayload {
    notes?: string | null;
    service_visit_id?: string | null;
}

export interface DeleteReceiptResponse {
    success: boolean;
}

File: backend/app/schemas/receipt.py
from pydantic import BaseModel, ConfigDict, Field
from datetime import datetime

class ReceiptBase(BaseModel):
    notes: str | None = Field(
        default=None,
        max_length=1000
    )

class ReceiptCreate(ReceiptBase):
    vehicle_id:         str
    service_visit_id:   str | None=None

class ReceiptUpdate(BaseModel):
    notes: str | None = Field(
        default=None,
        max_length=1000
    )
    service_visit_id: str | None = None

class ReceiptResponse(ReceiptBase):
    id:                 str
    user_id:            str
    vehicle_id:         str

    service_visit_id:   str | None = None

    original_filename:  str
    stored_filename:    str

    file_url:           str

    content_type:       str
    file_size:          int

    created_at:         datetime
    updated_at:         datetime

    model_config =      ConfigDict(from_attributes=True)